### PR TITLE
Add new `Style/EmptyLinesAroundExceptionHandlingKeywords` cop

### PR DESCRIPTION
**Pull request by pocke**
```ruby
 # bad
begin
  foo

rescue

  bar
end

 # good
begin
  foo
rescue
  bar
end
```

See also https://github.com/bbatsov/rubocop/pull/3984

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cops checks if empty lines exist around the bodies of begin
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb]: Add backquotes around keyword names.

File: lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      # sections. This cop doesn't check empty lines at begin body starting
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb]: at `begin` body beginning/end.

File: lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            keyword_locations_in_rescue(node)
+          when :ensure
+            keyword_locations_in_ensure(node)
+          else
+            []
+          end
+        end
+
+        def keyword_locations_in_rescue(node)
+          _, *resbodies, _ = *node
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb]: I'd use `_something` names here. They add more clarity IMO.

File: spec/rubocop/cop/style/empty_lines_around_exception_handling_keywords_spec.rb [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+rescue
+  f2
+else
+  f3
+end
+  CORRECTION
+  include_examples :offense,
+                   'rescue section ending for method definition',
+                   <<-CODE, <<-CORRECTION
+def foo
```
Comment (bbatsov) [inline spec/rubocop/cop/style/empty_lines_around_exception_handling_keywords_spec.rb]: I really hate outdented hereblocks. Use the `strip_margin` method from `powerpack` to pretty those up a bit.

File: lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            line = loc.line
+            keyword = loc.source
+            # on the keyword
+            check_line(style, line, format(MSG, 'on', keyword), &:empty?)
+            # under the keyword
+            check_line(style, line - 2, format(MSG, 'under', keyword), &:empty?)
+          end
+        end
+
+        def style
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb]: Is this a placeholder for making the cop configurable?

File: lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+        private
+
+        def check_body(node)
+          locations = keyword_locations(node)
+          locations.each do |loc|
+            line = loc.line
+            keyword = loc.source
+            # on the keyword
+            check_line(style, line, format(MSG, 'on', keyword), &:empty?)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb]: on -> before

File: lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+            # under the keyword
+            check_line(style, line - 2, format(MSG, 'under', keyword), &:empty?)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb]: under -> after

File: lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      #
+      #   rescue
+      #
+      #     do_something2
+      #   end
+      class EmptyLinesAroundExceptionHandlingKeywords < Cop
+        include EmptyLinesAroundBody
+        include OnMethodDef
+
+        MSG = 'Extra empty line detected %s the %s.'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_exception_handling_keywords.rb]: the %s -> `%s` (in backquotes)

**Comment by pocke** [comment]
> I really hate outdented hereblocks. Use the strip_margin method from powerpack to pretty those up a bit.

Thanks for the nice suggestion 👍 
`strip_indent`  method is used in the repository instead of `strip_margin`. 
So, I use the method instead. https://github.com/bbatsov/rubocop/search?utf8=%E2%9C%93&q=strip_indent




> Is this a placeholder for making the cop configurable?

No, I imitated here. https://github.com/bbatsov/rubocop/blob/ec3123fc3454b080e1100e35480c6466d1240fff/lib/rubocop/cop/style/empty_lines_around_method_body.rb#L35-L37
However, I think we can use the method for making configurable. I'll implement the configuration feature if there is a demand.


I've updated the other suggestions.

**Comment by bbatsov** [comment]
Looks good but has to be rebased.

**Comment by pocke** [comment]
I've rebased. Thanks.

**Pull request merged**